### ISSUE TITLE
Messages won't disappear when disappearing is enabled for group

### ISSUE THREAD

**Issue by GDB4J** [Auties00/WhatsappWeb4j#145]
Hi @Auties00 !

Great thanks for this API! Keep up the good work!
I'm stil figuring out a lot of things in your API.

Question:
When disappearing messages is enabled in a group chat, the messages actually don't dissapear :(

Left from every message (text, image, ...) there is a info icon, with this message:
![image](https://user-images.githubusercontent.com/108238720/175832027-0d325a88-b463-4e2d-9477-dbcd0bb1de9d.png)

I have also tried with configuring the version to the latest version by settings the Options:
`.version(new Version(2, 2222, 11))` (according to https://web.whatsapp.com/check-update?version=%s&platform=web)
Don't know if this relevant for this issue?
But nothing changed.

Is there a way to fix this?
Or is there a way to set an expiry date on a message?

My setup:
Spring boot 2.6.6
JDK 17.0.2
whatsappweb4j 3.0-RC11

Thanks for your time and effort in advance!
Kind regards,
G.

**Comment by Auties00** [comment]
> Hi @Auties00 !
> 
> Great thanks for this API! Keep up the good work! I'm stil figuring out a lot of things in your API.
> 
> Question: When disappearing messages is enabled in a group chat, the messages actually don't dissapear :(
> 
> Left from every message (text, image, ...) there is a info icon, with this message: ![image](https://user-images.githubusercontent.com/108238720/175832027-0d325a88-b463-4e2d-9477-dbcd0bb1de9d.png)
> 
> I have also tried with configuring the version to the latest version by settings the Options: `.version(new Version(2, 2222, 11))` (according to https://web.whatsapp.com/check-update?version=%s&platform=web) Don't know if this relevant for this issue? But nothing changed.
> 
> Is there a way to fix this? Or is there a way to set an expiry date on a message?
> 
> My setup: Spring boot 2.6.6 JDK 17.0.2 whatsappweb4j 3.0-RC11
> 
> Thanks for your time and effort in advance! Kind regards, G.

The easiest way is to wrap the message you want to send with MessageContainer.ofEphemeral(message). I'll make this automatic in the next update

**Comment by GDB4J** [comment]
@Auties00 
Seem's no solution so far:

doing this:

```
var message = TextMessage.newTextMessage() // Create a new text message
                .text("Test Message!") // Set the text of the message
                .create(); // Create the message
        api.sendMessage(chat, MessageContainer.ofEphemeral(message));
```

**Comment by Auties00** [comment]
> @Auties00 Seem's no solution so far:
> 
> doing this:
> 
> ```
> var message = TextMessage.newTextMessage() // Create a new text message
>                 .text("Test Message!") // Set the text of the message
>                 .create(); // Create the message
>         api.sendMessage(chat, MessageContainer.ofEphemeral(message));
> ```

That doesn't work?

**Comment by GDB4J** [comment]
Unfortunately not
![image](https://user-images.githubusercontent.com/108238720/175832779-629e305c-5941-4656-82c0-2326d36d4440.png)

**Comment by Auties00** [comment]
> Unfortunately not ![image](https://user-images.githubusercontent.com/108238720/175832779-629e305c-5941-4656-82c0-2326d36d4440.png)

So create a message info with your MessageContainer.ofEphemeral(...), create a context info with the property expiration set to a week in seconds or whatever is the timer for your chat, set the context in the message info and send

**Comment by GDB4J** [comment]
@Auties00 

Thanks this solved the problem indeed

Example:

```
        var message = TextMessage.newTextMessage() // Create a new text message
                .text("Test Message!") // Set the text of the message
                .contextInfo(new ContextInfo().expiration(60 * 60 * 24)) // Set expiration time to 24h (equals to your group setting)
                .create(); // Create the message
        api.sendMessage(chat, MessageContainer.ofEphemeral(message));
```

**Comment by Auties00** [comment]
> @Auties00
> 
> Thanks this solved the problem indeed
> 
> Example:
> 
> ```
>         var message = TextMessage.newTextMessage() // Create a new text message
>                 .text("Test Message!") // Set the text of the message
>                 .contextInfo(new ContextInfo().expiration(60 * 60 * 24)) // Set expiration time to 24h (equals to your group setting)
>                 .create(); // Create the message
>         api.sendMessage(chat, MessageContainer.ofEphemeral(message));
> ```

I'll keep it open because this should be done automatically. I'll fix it asap

**Comment by Auties00** [comment]
FIxed in 3.0-RC12

**Comment by GDB4J** [comment]
Hi @Auties00 ,

This is not working as default currently:

Using this piece of code to send a message to a group with dissapearingmode enabled:

```
var chat = api.store()
    .findChatByName("Chat_Group")
    .orElseThrow(() -> new NoSuchElementException("Chat does not exist!"));

var message = TextMessage.newTextMessage() // Create a new text message
        .text("Test Message!") // Set the text of the message
        .create(); // Create the message
api.sendMessage(chat, message);
```

The message is marked as "non deleteable" (like screenshots above).

Also when I'm trying to send like this:
`api.sendMessage(chat, MessageContainer.ofEphemeral(message));`
It's not working.


This is still needed to work correctly:
```
builder.contextInfo(new ContextInfo().ephemeralExpiration(60 * 60 * 24)) // Set expiration time to 24h (equals to group setting)
api.sendMessage(chat, MessageContainer.ofEphemeral(message));
```

I've tested this on version
` <version>3.0-RC13</version>`

Thanks in advance to take a look at it.
Kind regards

FYI: I can't reopen the issue
